documentation/docs/04-hooks.md

```diff
@@ -21,41 +21,23 @@ If unimplemented, defaults to `({ request, resolve }) => resolve(request)`.
 // type of string[] is only for set-cookie
 // everything else must be a type of string
 type ResponseHeaders = Record<string, string | string[]>;
-type RequestHeaders = Record<string, string>;
 
-export type RawBody = null | Uint8Array;
-
-type ParameterizedBody<Body = unknown> = Body extends FormData
-	? ReadOnlyFormData
-	: (string | RawBody | ReadOnlyFormData) & Body;
-
-export interface Request<Locals = Record<string, any>, Body = unknown> {
+export interface RequestEvent<Locals = Record<string, any>> {
+	request: Request;
```

Comment:
it seems funny to me that we have `url` and `request.url`. would it be possible to extend the request by adding `params` and `locals` to it to avoid this duplication?

I've always hated that the serverless environments call it an event. I think the idea is that you can have serverless endpoints which receive events that are not HTTP requests, but that's not applicable here. I'd love if we could find a way to avoid renaming `request` to `event` in all our docs

Reply:
`request.url` is a string, unfortunately, and not a `URL` instance. Rich and I just briefly chatted about this on Discord, and it seemed nicer to expose the parsed URL to users, since we are doing it anyway, and since it's reasonable to assume that they will want this data.

Extending official documented-in-MDN classes with additional fields doesn't sit well with me. I really don't think that's a good idea. I'm not wild about the name `event`, but am not sure what would be better.

Reply:
Extending `Request` is a non-starter — it'd be very confusing, mixing and matching stuff that's documented on MDN with our own monkey-patched properties, and in any case we don't own the request object (the adapters do). 

`request.url` is a string, whereas `url` is a `URL` object. An endpoint _could_ do this if they needed it...

```diff
-const foo = url.searchParams.get('foo');
+const foo = new URL(request.url).searchParams.get('foo');
```

...but since we're already generating the `URL` object ourselves, it probably makes sense to just expose it.

I don't love `event` either, but since we _do_ need to have an object with a `request` property (since monkey-patching is a no-go), and `request.request` sounds silly, I couldn't come up with anything better. Open to suggestions though